### PR TITLE
Handle max path limit on windows for schema names (backport)

### PR DESCRIPTION
**Pull request by mindaugasdirg**
Backport PR for #130 changes

### CODE REVIEW

File: packages/transformer/src/IModelTransformer.ts [hunk 1]
```diff
@@ -1322,7 +1322,9 @@ export class IModelTransformer extends IModelExportHandler {
     let schemaFileName = schema.name + ext;
     // many file systems have a max file-name/path-segment size of 255, so we workaround that on all systems
     const systemMaxPathSegmentSize = 255;
-    if (schemaFileName.length > systemMaxPathSegmentSize) {
+    // windows usually has a limit for the total path length of 256
+    const windowsMaxPathLimit = 260;
```
[the remark this answers is not in the document]
Comment (mindaugasdirg) [reply packages/transformer/src/IModelTransformer.ts:1326]: Forgot to update comment. I checked and microsoft documentation states that the path limit is 260. https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry

**Review by MichaelBelousov — APPROVED** [review]
approving since typescript change fixes linter issues when using typescript 5.2
We will want to bump our `@typescript-eslint` version eventually to be able to use new typescript

You may need to add an `// eslint-disable-next-line no-deprecation` comment to pass tests on some versions... we should consider not linting when doing minor version compatibility tests

**Review by MichaelBelousov — APPROVED** [review]

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -1,253 +1,344 @@
-lockfileVersion: 5.4
+lockfileVersion: '6.0'
```
Comment (MichaelBelousov) [inline pnpm-lock.yaml:1]: you installed with pnpm 8. This repository uses pnpm 7. Please do:
```sh
pnpm dlx pnpm@^7 install
# or 
npx pnpm@^7 install
```

**Review by MichaelBelousov — CHANGES_REQUESTED** [review]

File: pnpm-lock.yaml (hunk 2 above)
Comment (mindaugasdirg) [reply pnpm-lock.yaml]: Fixed it

**Review by MichaelBelousov — COMMENTED** [review]
LGTM but let's officially deprecate the resumption APIs if we disable the tests, 1.0 will remove them anyway

**Review by MichaelBelousov — APPROVED** [review]

**Pull request merged**